relayer/pkg/chainlink/txm/txm.go

```diff
@@ -287,17 +287,22 @@ func (txm *starktxm) broadcast(ctx context.Context, publicKey *felt.Felt, accoun
 		return txhash, fmt.Errorf("failed to create new account: %+w", err)
 	}
 
-	nonce, err := txm.nonce.NextSequence(publicKey)
-	if err != nil {
-		return txhash, fmt.Errorf("failed to get nonce: %+w", err)
+	unconfirmed := txm.txStore.GetSmallestUnconfirmedNonce(accountAddress)
+	txm.lggr.Debugw("estimate GetSmallestUnconfirmedNonce", "unconfirmed", unconfirmed)
+	if unconfirmed == nil {
+		unconfirmed, err = txm.nonce.NextSequence(publicKey)
+		if err != nil {
+			panic(err)
+		}
+		txm.lggr.Debugw("estimate NextSequence", "unconfirmed", unconfirmed)
 	}
 
 	tx := starknetrpc.InvokeTxnV3{
 		Type:          starknetrpc.TransactionType_Invoke,
 		SenderAddress: account.AccountAddress,
 		Version:       starknetrpc.TransactionV3,
 		Signature:     []*felt.Felt{},
-		Nonce:         nonce,
+		Nonce:         unconfirmed,
```

Comment:
won't the mempool reject this transaction because the nonce exists already in the mempool?